src/main/java/edu/kis/powp/jobs2d/drivers/WorkspaceManager.java

```diff
@@ -11,21 +11,35 @@
  * This class allows setting a {@link CanvaShape} to represent the workspace area,
  * and automatically renders its border using a special {@link Job2dDriver}.
  */
-public class WorkspaceManager {
+public class WorkspaceManager implements Job2dDriver {
     private CanvaShape currentCanvaShape;
     private final Job2dDriver borderDriver;
+    private boolean orClipeLine;
+    private static Job2dDriver innerDriver;
```

Comment:
Lack of SRP (low cohesion in this class). "Driver for avoiding exceeding device canvas" can be implemented as separate class (decorator pattern)